src/loader.js

```diff
@@ -67,14 +65,32 @@ export async function loader(imageBuffer) {
 	const generator = typeof generatorFactory === 'function'
 		? generatorFactory(options)
 		: new Generator(options);
+	const isExternalMode = generator.mode === 'external';
 	const imageSource = new Vinyl({
 		path: this.resourcePath,
-		contents: imageBuffer
+		contents: imageBuffer,
+		url: ''
 	});
 	const srcSet = [];
 
 	await attachMetadata(imageSource);
 
+	if (isExternalMode) {
+		const {
+			publicPath
+		} = getPaths(options, this, context, imageSource);
+		const {
+			protocol,
+			hostname
+		} = new URL(publicPath);
+
+		if (!protocol || !hostname) {
+			throw new Error('For external mode `publicPath` must be full URL with protocol and hostname');
+		}
+
+		imageSource.url = publicPath;
+	}
```

Comment:
```suggestion
	if (isExternalMode) {
		const {
			publicPath
		} = getPaths(options, this, context, imageSource);

		if (!publicPath || !/^https?:\/\//.test(publicPath)) {
			throw new Error('For external mode `publicPath` must be full URL with protocol and hostname');
		}

		imageSource.url = publicPath;
	}
```

it not possible to check url with `new URL`
```
new URL('/test')
VM149:1 Uncaught TypeError: Failed to construct 'URL': Invalid URL
    at <anonymous>:1:1
```

Reply:
Yep, because `/test` is not url, it's path. And we need a full url here, not path. So, it's correct to have error like `Invalid URL`.

Reply:
@someden, in this case, we should throw our error with detailed info

Reply:
ок